ledger-core/virtual/object/object.go

```diff
@@ -428,8 +435,63 @@ func (sm *SMObject) stepMigrate(ctx smachine.MigrationContext) smachine.StateUpd
 		return ctx.Error(fmt.Errorf("failed to publish state report"))
 	}
 
+	sdlCallSummarySync := ctx.Share(&sm.SummaryDone, 0)
+
+	if !ctx.Publish(callsummary.BuildSummarySyncKey(sm.Reference), sdlCallSummarySync) {
+		ctx.Log().Warn(struct {
+			*log.Msg  `txt:"failed to publish call summary sync key"`
+			Reference string
+		}{
+			Reference: sm.Reference.String(),
+		})
+		return ctx.Error(fmt.Errorf("failed to publish call summary sync key"))
+	}
+
+	return ctx.Jump(sm.stepPublishCallSummary)
+}
+
+func (sm *SMObject) stepPublishCallSummary(ctx smachine.ExecutionContext) smachine.StateUpdate {
+	pulseNumber := sm.pulseSlot.PulseData().PulseNumber
+	summarySharedStateAccessor, ok := callsummary.GetSummarySMSharedAccessor(ctx, pulseNumber)
+
+	if !ok {
+		ctx.InitChild(func(ctx smachine.ConstructionContext) smachine.StateMachine {
+			return callsummary.NewStateMachineCallSummary(pulseNumber)
+		})
+
+		summarySharedStateAccessor, _ = callsummary.GetSummarySMSharedAccessor(ctx, pulseNumber)
```

Comment:
check for `!ok` here

Reply:
```
ctx.InitChild(func(ctx smachine.ConstructionContext) smachine.StateMachine {
		p.initChildCtx(ctx, objectReference)

		return NewStateMachineObject(objectReference)
	})

	accessor, _ := p.TryGet(ctx, objectReference)
```

Example from catalog.go, should we change here too?

Reply:
si

Reply:
why I can not ignore ok here? Same as object catalog case

Reply:
Fixed + fixed in catalog.go